Nix maintenance

siraben opened the pull request:
- stdenv.lib has been deprecated (https://github.com/NixOS/nixpkgs/issues/108938)
- replace linuxkit-nix (unmaintained) with nix-docker
- updated niv sources

Eeems reviewed (changes requested):
I'm willing to still have nix builds as part of this repo, I'd prefer to deploy the nix package, and only run it on the master branch.

Please rework this branch to handle that.

siraben commented:
@Eeems Ok, I rebased this branch on top of master and fixed the Nix build. I'm not sure how to rewrite the workflow as you specified, could you clarify what the uploaded artifact should be?

Eeems commented:
It should just be whatever nix uses for a package file that can be installed with nix on device.

Eeems commented on oxide.nix:
```diff
@@ -14,17 +14,38 @@ stdenv.mkDerivation {
 
   installPhase = ''
-    cp -r release/. $out
+    install -D -m 755 -t $out/opt/bin release/opt/bin/erode
```
This looks like it's going to be painful to maintain. Is there a reason you made it explicit instead of just using the entire contents of the release folder?

Eeems replied:
@siraben poke?
